src/esn.calendar.libs/app/components/time-select/time-select.controller.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+require('../../services/fc-moment.js');
+
+angular.module('esn.calendar.libs')
+  .controller('calTimeSelectController', calTimeSelectController);
+
+function calTimeSelectController($scope, $element, calMoment) {
+  const self = this;
+
+  self.$onInit = onInit;
+  self.onTimeInputChange = onTimeInputChange;
+  self.onSelectingTimeOption = onSelectingTimeOption;
+  self.onTimeSelectBlur = onTimeSelectBlur;
+  self.onInputKeydown = onInputKeydown;
+
+  function onInit() {
+    self.timeOptions = getTimeOptions();
+    self.isInputValid = true;
+    self.selectedTime = this.date.locale(self.locale).format(self.timeFormat);
+
+    // $watch for handling the offset being added to the date
+    $scope.$watch('ctrl.date', function(newDate) {
+      initTimeInput(newDate);
+    });
+
+    // The input field is added to the click refs so that the dropdwon doesn't close when clicking inside the input
+    $scope.additionalClickRefs = [$element.find('input')[0]];
+  }
+
+  function onInputKeydown(event, mdMenu) {
+    if (event.keyCode !== 13) return;
```

Comment:
```suggestion
    if (event.key === 'Enter') return;
```

Sorry but let's use this instead. `event.keyCode` is deprecated as I've just read [here](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode).